utility/serial.cpp

```diff
@@ -92,6 +92,71 @@ void ArdusatSerial::begin(unsigned long baud, bool setXbeeSpeed)
   }
 }
 
+/**
+ * Begin serial communications with a Sparkfun BlueSMiRF module at the specified baud
+ * rate.
+ *
+ * Note that baud rates above ~57600 are not well-supported by SoftwareSerial, and 
+ * even 57600 may cause some bugs.
+ *
+ * @param baud rate for serial communications
+ *
+ * NOTE: If this function is called multiple times with different values for 'baud'
+ *       without losing power in between calls, the baud rate will not be updated
+ *       after the first call.
+ */
+void ArdusatSerial::beginBluetooth(unsigned long baud)
+{
+  bool validBaud = true;
+  char baudCmd[8];
```

Comment:
this is actually causing a buffer overrun. Remember, in C, strings are null-terminated, so every string has an implicit '\0' after the visible characters. strcpy copies every character in the source string over, including that terminating character, so your calls to strcpy below are writing 9 characters, not 8. That means they're setting the memory address &baudCmd + 9 = '\0';, which may not mess with anything immediately, but is not good/predictable.

Given that we know that there's another (at present unused) cmd option for 115200 baud which would have another character to it, I'd probably make this a char [10]; just in case we add the command in the future and whoever does that forgets to check the buffer length. An extra byte allocated on the stack for this function call isn't a big deal.